bert/README.md

```diff
@@ -0,0 +1,244 @@
+* [Domain-Specific Language Model Pretraining for Biomedical Natural Language Processing](https://arxiv.org/abs/2007.15779)
+* [LinkBERT: Pretraining Language Models with Document Links](https://arxiv.org/abs/2203.15827)
+
+In addition to being able to take advantage of pre-training on in-domain data, training from scratch means that you control the data start to finish. Publicly available pre-training corpuses cannot be used in many commercial cases due to legal considerations. Our starter code can easily be modified to handle custom datasets beyond the C4 example we provide.
+
+One may wonder, why start from scratch when public data isn't a concern? Granted that it is better to train on domain-specific data, can't that happen as "domain adaptation" from a pre-trained checkpoint? There are two reasons not to do this, one theoretical and one practical. The theory says that, because we are doing non-convex optimization, domain adaptation "may not be able to completely undo suboptimal initialization from the general-domain language model" [(Gu et al., 2020)](https://arxiv.org/abs/2007.15779).
+
+The practical reason is that certain outcomes are only available if the model and tokenizer are pre-trained from scratch.
+
+So, for example, if you want to use ALiBi positional embeddings (and [you probably should](https://ofir.io/The-Use-Case-for-Relative-Position-Embeddings/), they seem to improve LM perplexity, downstream accuracy, and allow the model to generalize to longer sequences than seen at train time), you need to train from scratch (or fine-tune from a checkpoint which was pre-trained with ALiBi positional embeddings, which we will be releasing!). Or if you want to use Gated Linear Units in your feedforward layers ([as recommended by Noam Shazeer](https://arxiv.org/abs/2002.05202), one of the authors of the original Transformers paper), again, you have to train with them from scratch.
+
+Another good example is domain-specific tokenization. In the biomedical domain, words may be split by the pre-trained BERT tokenizer in ways that make downstream tasks more difficult and computationally expensive. For example, the common drug "naloxone" in tokenized by `bert-base-uncased` tokenizer into the 4 tokens `[na, ##lo, ##xon, ##e]` [(Gu et al., 2020)](https://arxiv.org/abs/2007.15779), making tasks like NER more difficult and using more of the limited sequence length available.
+
+Now that we've convinced you that you should train a Mosaic BERT from scratch, let's get into the how :) 
+
+# Setup
+
+Here's what you need to get started with our BERT implementation:
+
+* Use a Docker image with PyTorch 1.12+, e.g. [MosaicML's PyTorch base image](https://hub.docker.com/r/mosaicml/pytorch/tags)
+  * Recommended tag: `mosaicml/pytorch:1.12.1_cu116-python3.9-ubuntu20.04`
+  * This image comes pre-configured with the following dependencies:
+    * PyTorch Version: 1.12.1
+    * CUDA Version: 11.6
+    * Python Version: 3.9
+    * Ubuntu Version: 20.04
```

Comment:
Can we turn this into just `docker pull mosaicml/pytorch:1.12.1_cu116-python3.9-ubuntu20.04` + a link to more info about what's in this image if you're curious (maybe at the bottom of this readme, but ideally shared with other benchmarks since they all have this).

Trying to get this section down to a few lines the user can copy-paste to get started. Something like:

```
git clone https://github.com/mosaicml/benchmarks.git
cd bert
docker pull mosaicml/pytorch:1.12.1_cu116-python3.9-ubuntu20.04
...
```

Alternatively, we could have something like this in a Quickstart section at the top and this could be the more detailed version we link to.